sql/core/src/main/scala/org/apache/spark/sql/execution/Generate.scala

```diff
@@ -34,24 +35,12 @@ import org.apache.spark.sql.catalyst.expressions._
 @DeveloperApi
 case class Generate(
     generator: Generator,
+    output: Seq[Attribute],
```

Comment:
Okay, but that sounds like an argument to take in only the produced attributes as a parameter and have `output` be a def that is calculated based on the value of `join`, etc.  I don't see why we should keep the unused, complicated, redundant `Attribute`s in generator (actually they are used once in this file, but that seems like any easy way to allow these two things to get out of sync)